factorial/factorial.c

```diff
@@ -0,0 +1,45 @@
+#include <inttypes.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+// Signatures of functions defined later
+int validateInput(int, char **);
+uintmax_t factorial(uintmax_t);
+
+// Entrypoint
+int main( int argc, char *argv[] )  {
+    uintmax_t value = validateInput(argc, argv);
+    printf("Input value: %lu\n", value);
+    uintmax_t result = factorial(value);
+    printf("     Result: %lu\n", result);
+    return 0;
+}
+
+// Calculate factorial
+uintmax_t factorial(uintmax_t value) {
+    if (value < 2) {
+        return 1;
+    }
+    uintmax_t counter = value;
+    uintmax_t result = value;
+    while (counter > 1) {
+        result = result * (counter - 1);
```

Comment:
why `result * (counter - 1)` and not just `result * counter`?

I think you can simplify the code by setting up your initial conditions a bit differently.

Reply:
Agreed.  Will fix.

Reply:
fixed